tests/api/test_posts.py

```diff
@@ -0,0 +1,59 @@
+from typing import Dict, Iterable
+
+import aiohttp
+import pytest
+from aleph_message.models import MessageType
+
+from .utils import get_messages_by_keys
+
+POSTS_URI = "/api/v0/posts.json"
+
+
+def assert_posts_equal(posts: Iterable[Dict], expected_messages: Iterable[Dict]):
+    posts_by_hash = {post["item_hash"]: post for post in posts}
+
+    for expected_message in expected_messages:
+        post = posts_by_hash[expected_message["item_hash"]]
+        assert "_id" not in post
+
+        assert post["chain"] == expected_message["chain"]
+        assert post["channel"] == expected_message["channel"]
+        assert post["sender"] == expected_message["sender"]
+        assert post["signature"] == expected_message["signature"]
+
+        if expected_message.get("forgotten_by", []):
+            assert post["content"] is None
+            continue
+
+        if "content" not in expected_message["content"]:
+            # TODO: there is a problem with the spec of posts: they can be specified
+            #       without an internal "content" field, which does not break the
+            #       endpoint but returns the content of message["content"] instead.
+            #       We skip the issue for now.
+            continue
+
+        assert post["content"] == expected_message["content"]["content"]
+
+
+async def get_posts(api_client, **params) -> aiohttp.ClientResponse:
+    return await api_client.get(POSTS_URI, params=params)
+
+
+async def get_posts_expect_success(api_client, **params):
+    response = await get_posts(api_client, **params)
+    assert response.status == 200, await response.text()
```

Comment:
Why not use something like [response.raise_for_status()](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.raise_for_status) instead ?

Reply:
Because 'raise_for_status()' does not print the response body, which I want to happen in case of error.